aws/src/main/java/org/apache/iceberg/aws/s3/signer/S3V4RestSignerClient.java

```diff
@@ -328,6 +333,27 @@ public SdkHttpFullRequest sign(
     return mutableRequest.build();
   }
 
+  /**
+   * Only add body for DeleteObjectsRequest. Refer to
+   * https://docs.aws.amazon.com/AmazonS3/latest/API/API_DeleteObjects.html#API_DeleteObjects_RequestSyntax
+   */
+  private String bodyAsString(SdkHttpFullRequest request) {
+    if (isDeleteObjectsRequest(request) && request.contentStreamProvider().isPresent()) {
+      try (InputStream is = request.contentStreamProvider().get().newStream()) {
+        return IoUtils.toUtf8String(is);
+      } catch (IOException e) {
+        LOG.debug("Failed to set body for S3 sign request", e);
```

Comment:
nit: technically we're not setting the body for the S3 sign request here, but just getting the body from the request as a string

Reply:
Good point, I've edited the log to be "Failed to determine body for S3 sign request"